libvast/src/system/index.cpp

```diff
@@ -342,7 +347,9 @@ void index_state::decomission_active_partition() {
   auto part_dir = partition_path(id);
   auto synopsis_dir = partition_synopsis_path(id);
   VAST_DEBUG("{} persists active partition to {}", self, part_dir);
-  self->request(actor, caf::infinite, atom::persist_v, part_dir, synopsis_dir)
+  self
+    ->request(actor, caf::infinite, atom::persist_v,
+              vast::path{part_dir.string()}, vast::path{synopsis_dir.string()})
```

Comment:
Note: If you want to change these, you need to adapt the `active_partition_actor` type definition in `vast/system/actors.hpp`, and the corresponding lambdas in the `active_partition` function definition return value.

Reply:
Yeah, I went down that path for a bit this morning, but it seemed a bit more nuanced than that. I'd have to add `CAF_ADD_TYPE_ID` for `std::filesystem::path` and even when I did that and updated the callers appropriately (or so I think), there were still caf errors, which are pretty poor IMO. I'm hoping to change the actors last once all of the surrounding code doesn't use `vast::path` anymore. What do you think? It might be something good for us to pair on when we get to that point.

Reply:
The type ID part is the worst, and we struggle with its requirements all the time. We'll have to think about possibly forward-declaring `std::filesystem::path`, splitting our type ID blocks into multiple (again), or [adding a `caf::inspector_access<std::filesystem::path>` specialization](https://actor-framework.readthedocs.io/en/stable/TypeInspection.html#specializing-inspector-access).

After giving this some thought, I feel like we either pair as you suggest or we do it ourselves internally sometime over the coming weeks.

Reply:
I've implemented this in https://github.com/tenzir/vast/pull/1508

Reply:
> I've implemented this in #1508

LGTM.  I'll try to incorporate it into the actors likely in a follow-up PR.